ItemLevelDownloader/Engine/logger.ts

```diff
@@ -0,0 +1,69 @@
+import { ItemType } from '../Models';
+import { TicketState } from '../Models/ticketState';
+import { ArtifactItemStore } from '../Store/artifactItemStore';
+
+export class Logger {
+
+    constructor(verbose: boolean) {
+        this.verbose = verbose
+    }
+
+    public logInfo(message: string) {
+        if (this.verbose) {
+            console.info(message);
+        }
+    }
+
+    public logError(message: string) {
+        console.error(message);
+    }
+
+    public logMessage(message: string) {
+        console.log(message);
+    }
+
+    public logSummary(store: ArtifactItemStore): void {
+        var tickets = store.getTickets();
+        tickets = tickets.filter(x => x.artifactItem.itemType == ItemType.File);
+
+        var processedItems = tickets.filter(x => x.state == TicketState.Processed);
+        var queuedItems = tickets.filter(x => x.state == TicketState.InQueue);
+        var skippedItems = tickets.filter(x => x.state == TicketState.Skipped);
+        var failedItems = tickets.filter(x => x.state == TicketState.Failed);
+
+        console.log(
+            "Total: " + tickets.length
+            + ", Processed: " + processedItems.length
+            + ", Queued: " + queuedItems.length
+            + ", Skipped: " + skippedItems.length
+            + ", Failed: " + failedItems.length);
+        
+        console.log("Summary:");
```

Comment:
this should go within if (verbose) block? or did you intend to have a shorter summary in case logging is not verbose?

Reply:
yes will move to verbose block.